index.js

```diff
@@ -25,6 +42,21 @@ function isV1Addon(pkg) {
   return false;
 }
 
+function classicAppPath(pkgRoot, importPath) {
+  if (importPath.startsWith('config/') || importPath.startsWith('tests/')) {
+    return `${pkgRoot}/${importPath}`;
+  }
+
+  return `${pkgRoot}/app/${importPath}`;
+}
+
+function classicAddonPath(pkgRoot, importPath) {
+  if (importPath.startsWith('test-support/')) {
+    return `${pkgRoot}/addon-test-support/${importPath.substring(13)}`;
+  }
+  return `${pkgRoot}/addon/${importPath}`;
+}
```

Comment:
I think technically `addon-name/config/environment.js` is also a valid import within an engine. Might want to add this as a limitation though, since its only really valid at the engine level.

Reply:
Should we just add support for it?  Seems extremely trivial to do